src/ocamlorg_package/lib/ocamlorg_package.ml

```diff
@@ -367,15 +367,12 @@ let doc_exists t name version =
 
 let latest_documented_version t name =
   let rec aux vlist =
-    let open Lwt.Syntax in
+    let open Lwt.Infix in
     match vlist with
     | [] -> Lwt.return None
-    | _ -> (
-        let version = List.hd vlist in
-        let* doc = doc_exists t name version in
-        match doc with
-        | Some version -> Lwt.return (Some version)
-        | None -> aux (List.tl vlist))
+    | version :: vlist ->
+        doc_exists t name version
+        >>= Option.fold ~none:(aux vlist) ~some:Lwt.return_some
```

Comment:
I know it's a thing people commonly do... but maybe we can try not to reuse/overshadow parameter names in match-expressions?

```suggestion
    | version :: vs ->
        doc_exists t name version
        >>= Option.fold ~none:(aux vs) ~some:Lwt.return_some
```

Reply:
Also not sure if we want to keep the code more verbose and obvious, to make it easier for beginners to contribute. :thinking: